Library/Homebrew/utils/github.rb

```diff
@@ -759,7 +759,24 @@ def create_bump_pr(info, args:)
     end
   end
 
-  def pull_request_commits(user, repo, pr)
-    open_api(url_to("repos", user, repo, "pulls", pr, "commits?per_page=100")).map { |c| c["sha"] }
+  def pull_request_commits(user, repo, pr, per_page: 100)
+    pr_data = open_api(url_to("repos", user, repo, "pulls", pr))
+    commits_api = pr_data["commits_url"]
+    commit_count = pr_data["commits"]
+    page = 0
+    commits = []
+
+    loop do
+      page += 1
+      result = open_api(commits_api + "?per_page=#{per_page}&page=#{page}")
+      commits.concat(result.map { |c| c["sha"] })
+      break if result.empty? || page * per_page >= commit_count
```

Comment:
The potential for this to be infinite concerns me. Could we set a maximum upper bound here?

Reply:
Well, the API request limit will kick in at 49 pages and blow up the function call anyway. Is that an acceptable upper bound?

Reply:
Yeh! Could that be hardcoded on our end, too, so we're not reliant on GitHub doing what they say they will 😅

Reply:
Ok, how about raising an error if we hit 50 pages, or the number of commits to be pulled is greater than 5000

Reply:
Sounds good 👍🏻 A numbered loop with an early `break` may make sense over a `loop` in that case.